common/llguidance.cpp

```diff
@@ -0,0 +1,273 @@
+#include "common.h"
+#include "sampling.h"
+#include "log.h"
+#include "llama.h"
+
+#include <cmath>
```

Comment:
Remove unused headers:

```suggestion
#include "sampling.h"
#include "log.h"
#include "llama.h"
```

Reply:
`<cmath>` is actually required on Linux for `INFINITY` constant; I did remove common.h though, thank  you!